docs/cudf/source/user_guide/pandas-comparison.md

```diff
@@ -82,9 +82,16 @@ using `.from_arrow()` or `.from_pandas()`.
 
 ## Result ordering
 
-By default, `join` (or `merge`) and `groupby` operations in cuDF
-do *not* guarantee output ordering.
-Compare the results obtained from Pandas and cuDF below:
+In Pandas, `join` and `groupby` operations provide certain guarantees
+about the order of rows in the result returned.  In Pandas `join`, the
+order of join keys are either preserved or sorted lexicographically by
+default.  `groupby` sorts the group keys, and preserves the order of
+rows within each group. In some cases, disabling this option in Pandas
+can yield better performance.
+
+By contrast, cuDF's default behavior is to return rows in a
+non-deterministic order to maximize performance.  Compare the results
+obtained from Pandas and cuDF below:
```

Comment:
Maybe note that for a groupby _within_ a group the ordering of the input is preserved?

Reply:
Do we _need_ to make that guarantee? Internally, we do use a stable sort when doing a sort-based groupby but can we leave that as an implementation detail?

Reply:
Oh probably not, but I don't know the details of what we guarantee as an API statement, and what we provide as an implementation detail. So let us leave it as is here.